File: spring/member-server/src/main/java/com/example/memberserver/member/MemberDto.java
package com.example.memberserver.member;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
public class MemberDto {

    private String name;
    private String email;
    private String password;

    public static MemberDto of(Member member) {
        MemberDto memberDto = new MemberDto();
        memberDto.setName(member.getName());
        memberDto.setEmail(member.getEmail());
        return memberDto;
    }
}

File: user-server-firebase-spring-kotlin/src/main/kotlin/com/example/userserverfirebasespringkotlin/MemberDto.kt
package com.example.userserverfirebasespringkotlin

data class MemberDto(
    val name: String = "",
    val email: String = "",
    val password: String = "",
    val permissionIds: List<String> = listOf(),
    val permissionNames: List<String> = listOf(),
) {

    companion object {
        fun of(member: Member): MemberDto {
            return MemberDto(
                name = member.name,
                email = member.email,
                permissionNames = member.permissions.map { it.name }
            )
        }
    }
}
